internal/storage/payload_writer.go

```diff
@@ -45,17 +45,24 @@ type NativePayloadWriter struct {
 	flushedRows int
 	output      *bytes.Buffer
 	releaseOnce sync.Once
+	dim         int
 }
 
 func NewPayloadWriter(colType schemapb.DataType, dim ...int) (PayloadWriterInterface, error) {
 	var arrowType arrow.DataType
+	var dimension int
 	if typeutil.IsVectorType(colType) {
 		if len(dim) != 1 {
-			return nil, fmt.Errorf("incorrect input numbers")
+			return nil, merr.WrapErrParameterInvalid("pass in dim", "not", "incorrect input numbers")
 		}
 		arrowType = milvusDataTypeToArrowType(colType, dim[0])
+		dimension = dim[0]
 	} else {
+		if len(dim) != 0 {
```

Comment:
behavior changed, as it works well if passed `[1]` as the `dim` parameter, but after this not